### PR TITLE
ID-291 pre-requisite for adding scala-steward

### PR DESCRIPTION
**Pull request by andy7i**
https://broadworkbench.atlassian.net/browse/ID-291

Set up scala steward so that dependency updates are a bit more streamlined and get all that engoodeningness.

Docker tag names are based on the github branch name. Sometimes github branch names contain slashes (/) and this causes issues for the docker tag, due to the [docker tag name restrictions](https://docs.docker.com/engine/reference/commandline/tag/#:~:text=A%20tag%20name%20must%20be,a%20maximum%20of%20128%20characters.)

Copied over other files based on directions in https://github.com/broadinstitute/scala-steward/blob/main/README.md#set-up-your-repo and added the [@scala-steward Team](https://github.com/orgs/broadinstitute/teams/scala-steward/members) as a Writer

Related PR: https://github.com/broadinstitute/scala-steward/pull/9

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: .github/workflows/merge_scalasteward_prs.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: dependency-bot
```
Comment (andy7i) [inline .github/workflows/merge_scalasteward_prs.yml:1]: copied from https://github.com/broadinstitute/sam/blob/develop/.github/workflows/merge_scalasteward_prs.yml

File: .scala-steward.conf [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Repository-specific configuration
```
Comment (andy7i) [inline .scala-steward.conf:1]: copied from https://github.com/broadinstitute/sam/blob/develop/.scala-steward.conf

File: docker/build.sh [hunk 3]
```diff
@@ -23,6 +23,9 @@ set -e
 # Set default variables
 DOCKER_CMD=
 BRANCH=${BRANCH:-$(git rev-parse --abbrev-ref HEAD)}  # default to current branch
+REGEX_TO_REPLACE_ILLEGAL_CHARACTERS_WITH_DASHES="s/[^a-zA-Z0-9_.\-]/-/g"
```
Comment (andy7i) [inline docker/build.sh:26]: these changes were copied from https://github.com/broadinstitute/sam/pull/672/files

File: .github/workflows/merge_scalasteward_prs.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+on:
+  workflow_dispatch:
+  # Allows manually triggering of workflow on a selected branch via the GitHub Actions tab.
+  # GitHub blog demo: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/.
+
+#  push:
+#    branches:
+#      - workflow
```
Comment (gpolumbo-broad) [inline .github/workflows/merge_scalasteward_prs.yml:10]: can we delete commented out section?

File: docker/build.sh (hunk 3 above)
Comment (gpolumbo-broad) [reply docker/build.sh:26]: 😂

**Review by gpolumbo-broad — APPROVED** [review]

**Review by tlangs — APPROVED** [review]

File: .github/workflows/merge_scalasteward_prs.yml (hunk 4 above)
Comment (andy7i) [reply .github/workflows/merge_scalasteward_prs.yml:10]: I think it's useful to keep this in. It's a bit of documentation for how to enable a useful feature.

**Pull request merged**